mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -692,6 +744,29 @@ class SparseVector @Since("1.0.0") (
 
   private[spark] override def toBreeze: BV[Double] = new BSV[Double](indices, values, size)
 
+  private[spark] override def mapActiveValues(f: Double => Double): Vector =
+    new SparseVector(size, indices.clone(), values.map(f))
+
+  private[spark] def mapActivePairs(f: (Int, Double) => Double): Vector = {
```

Comment:
I am learning Scala. Could you please explain how this maps only the active pairs? 
what is happening in the line no: 756